src/apps/experimental/features/details/hooks/api/useGetItemByType.ts

```diff
@@ -0,0 +1,61 @@
+import type { AxiosRequestConfig } from 'axios';
+import { getUserLibraryApi } from '@jellyfin/sdk/lib/utils/api/user-library-api';
+import { getLiveTvApi } from '@jellyfin/sdk/lib/utils/api/live-tv-api';
+import { useQuery } from '@tanstack/react-query';
+import { type JellyfinApiContext, useApi } from 'hooks/useApi';
+import type { ItemDto } from 'types/base/models/item-dto';
+import { ItemKind } from 'types/base/models/item-kind';
+
+const getItemByType = async (
+    apiContext: JellyfinApiContext,
+    itemType: ItemKind,
+    itemId: string,
+    options?: AxiosRequestConfig
+) => {
+    const { api, user } = apiContext;
+    if (!api) throw new Error('No API instance available');
```

Comment:
Even though it _shouldn't_ be possible with the enabled check, I have found I sometimes still encounter this while running webpack dev server. I have started just logging a warning instead of throwing, but maybe we should throw so we are forced to handle the error state properly. I'm curious what you think about this.

Reply:
can you give me more info 😟

Reply:
I've never been able to replicate it reliably, but here is an example of where I just logged a warning instead of threw an error: https://github.com/jellyfin/jellyfin-web/blob/master/src/apps/dashboard/features/activity/api/useLogEntries.ts#L14-L17

Reply:
@thornbill we can use this approach too to display error on the page
```js
if (!api) return Promise.reject(new Error('[getItemByType] No API instance available'));
```

Reply:
That looks like a good option!